action/protocol/staking/protocol.go

```diff
@@ -261,13 +262,17 @@ func (p *Protocol) handle(ctx context.Context, act action.Action, csm CandidateS
 		return nil, nil
 	}
 
+	logs := []*action.Log{rLog.Build(ctx, err)}
 	if err == nil {
-		return p.settleAction(ctx, csm, uint64(iotextypes.ReceiptStatus_Success), rLog.Build(ctx, err))
+		if aLog != nil && aLog.IsWithdrawBucket() {
```

Comment:
if aLog is not nil, why we still want to check its type?

Reply:
fixed